### ISSUE TITLE
Application volume adjuster: change the control for enabling / disabling in the audio settings pannel to a checkbox

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#17122]
### Steps to reproduce:
1. Open the audio pannel with nvda+ctrl+u
2. Navigate with tab to the applications volume adjuster combo box
3. Explore its values with up and down arrow keys.

### Actual behavior:
There is inconsistency with options that have only two possible values (enabled or disabled" which are not part of advanced settings pannel.
In NVDA's GUI design terms, the intuitive way is to have a checkbox in such cases, otherwise users get confused and the GUI appears messy.

### Expected behavior:
Change the combo box of this option to a check box which can be enabled or disabled (disabled by default).

### NVDA logs, crash dumps and other attachments:
n/a

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
last alpha

#### Windows version:
Windows 11 23 H2

#### Name and version of other software in use when reproducing the issue:
n/a

#### Other information about your system:
n/a

#### Does the issue still occur after restarting your computer?
n/a

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
n/a

#### If NVDA add-ons are disabled, is your problem still occurring?
n/a

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
n/a

**Comment by CyrilleB79** [comment]
@Adriani90, seems we have already had this discussion somewhere else...

In NVDA, there are options with 3 values: Enabled, Disabled and Default. You seem to think that this type of options is reserved to Advanced settings panel. But currently that's not the case.

In addition to Volume adjuster, we have:
* Unicode normalization (in Speech settings)
* Show selection (in Braille settings)
* Unicode normalization (in Braille settings)

So this type of 3-value option is not currently reserved to experimental features in Advanced settings panel. It's used for any 2-value option whose default is likely to change in the future.

I acknowledge that from a UX point of view, a checkbox in the GUI is nicer than these combo-boxes. But that's not specific to the Volume adjuster one.

Would you like to broaden this issue to any non-experimental option (i.e. not in Advanced settings)? If yes, I'd suggest that you suggest a proposal (or more) to be allowed to change easily the default value of an option in the next releases.

**Comment by seanbudd** [comment]
Closing, this is the intended use of feature flags

**Comment by Adriani90** [comment]
This feature flag thing makes sense for settings with more than two values, but really not for just "enabled" or "disabled". But yeah, it seems discussion is blocked on this topic everytime and I am not sure what the reason behind it is. I will not start a discussion again on this. Sooner or later it will be perceived as inconsistent the more such combo boxes will come to the GUI anyway.

**Comment by SaschaCowley** [comment]
@Adriani90, as explained by @CyrilleB79 above, the purpose of feature flags is to allow us to have settings that will likely change their default value in future. This is particularly important for features that are not "advanced", but which are somewhat experimental or in a state of flux. Having a default option as well as explicit enabled and disabled options allows users to explicitly opt in or out of the feature, or to allow NVDA to decide. If you have a proposal for a more elegant way of handling this use-case, by all means share it, but do keep in mind that users should:
- Be able to explicitly opt in or out of the setting, and not have a future update override their selection; and
- Return from having explicitly opted in or out, to being in the default state where a future update may change the current value.

**Comment by Adriani90** [comment]
@SaschaCowley thank you for the elaboration, I understand the use case for this when there are multiple values other than just completely enabled or completely disabled (e.g. as for registering for UIA events, or using UIA for MS Word). But for enabled or disabled values, changing the standard default state from NVDA side for the user it feels the same like having a checkbox which you enabled and after an update it is disabled.
In the user guide it is not clear at all that users can opt in or out explicitely for an option or that theere is the option to let NVDA developers decide.

Actually the main motivation for the feature flag and the default options was originally to let developers define standard behaviors that are not exactly similar to the options the user can opt for explicitely, but maybe people can proove me wrong. That's why it was actually added to NVDA for experimental features in the advanced settings pannel and not to other settings pannels. There is no feature out there at all that really made use of the feature flag as it was meant to be. All default values in the feature flags have a exactly similar behavior to one of the option the user can explicitely opt for.

As with regard to settings which only have compeltely enabled or completely disabled options, a checkbox has the advange that you can add accelerators to it and users can use them to enable or disable settings in the coresponding settings pannels. But maybe that is possible for combo box values as well?

**Comment by SaschaCowley** [comment]
> But for enabled or disabled values, changing the standard default state from NVDA side for the user it feels the same like having a checkbox which you enabled and after an update it is disabled.

I don't think we would usually do this on a stable version unless having the feature enabled introduced a serious issue. 

> As with regard to settings which only have compeltely enabled or completely disabled options, a checkbox has the advange that you can add accelerators to it and users can use them to enable or disable settings in the coresponding settings pannels. But maybe that is possible for combo box values as well?

Not as far as I am aware.

I am not entirely sure what you are getting at with the rest of your message.

**Comment by Adriani90** [comment]
> I don't think we would usually do this on a stable version unless having the feature enabled introduced a serious issue.



I agree, though such a change would go to alpha first anyway. That happened with unicode normalization or with cancelable speech setting as well. If these settings were checkboxes instead of combo boxes with a default option  item, the  effect for the user is exactly the same except in those cases a checkbox has more advantages for the user. My concern is that the default value item in the combo box is in fact just a duplicate item for an option that the user can choose in that combo box. And that‘s not how these default option items were originally meant to be.

> I am not entirely sure what you are getting at with the rest of your message.
 

Ad

let‘s say you develop a feature wich can be enabled or disabled, and you want to experiment what happens when a partially enabled state does e.g. for users of a certain version of Windows, then the enabled and disabled options are the options are those which allow the users  to explicitely opt in or out. The default value though would be the state that you as a developer try to experiment with or test. It could be partially enabled e.g. only in certain cases, or you can change it to be default enabled only on windows 10 or what ever. So the default option is not really a very fix option. And ideally  should not be a duplicate of other options in the combo box. At least this is my understanding.

**Comment by Adriani90** [comment]
And if the default option item is for test / experiment purposes, then it doesn‘t have to appear in settings pannels other than advanced settings pannel which is exactly the right place for experimental features.
